### PR TITLE
Updated Cloud FAQ

### PR DESCRIPTION
**Pull request by russjones**
Minor cleanup for Cloud FAQ.

* Self-serve SOC 2 download
* Clarified upgrade cadence and production readiness

### CODE REVIEW

File: docs/pages/choose-an-edition/teleport-cloud/faq.mdx [hunk 1]
```diff
@@ -139,9 +139,18 @@ Currently there is no option to select an alternative region for storage.
 
 ## Will Teleport be automatically upgraded with each release?
 
-The Teleport control plane is routinely upgraded within 1-2 weeks of a Teleport Core or Enterprise Release.
-New features may not be available to cloud users until compatibility is tested and fixed, and some features may not work in the cloud environment.
-Individual customers may not be upgraded to the latest release if they are running agents that are outside of the supported compatibility of the control plane.
+Teleport Cloud follows the Teleport [Production
+readiness](../../preview/upcoming-releases.mdx) guidelines.
+
+- Teleport Cloud control plane major version upgrades occur after the first
```
Comment (zmb3) [inline docs/pages/choose-an-edition/teleport-cloud/faq.mdx:145]: This is the only place in the entire docs where we refer to the Teleport Cloud _control plane_. If we want to use a new term we should probably define it first.

Alternatively, just say Auth Servers and Proxy Servers.

File: docs/pages/choose-an-edition/teleport-cloud/faq.mdx [hunk 2]
```diff
@@ -139,9 +139,18 @@ Currently there is no option to select an alternative region for storage.
...
+  minor release (for example (=teleport.major_version=).1.0). This typically
+  happens within one month of a new major version release becoming available.
+- After the control plane has been upgraded to the first major version, minor
+  and patch release upgrades typically occur within a week.
+
+Individual customers may not be upgraded to the latest release if they are
```
Comment (zmb3) [inline docs/pages/choose-an-edition/teleport-cloud/faq.mdx:151]: Are you sure we want to add this. Some may interpret as permission to never upgrade and expect that we will indefinitely hold back their control plane for them.

**Review by zmb3 — APPROVED** [review]

File: docs/pages/choose-an-edition/teleport-cloud/faq.mdx (hunk 2 above)
Comment (russjones) [reply docs/pages/choose-an-edition/teleport-cloud/faq.mdx]: Good point, removed it because we're soon not even going to offer this.

**Review by jimbishopp — APPROVED** [review]

**Comment by alexfornuto** [comment]
Any reason not to backport this beyond v12?

**Pull request merged**